src/emqttd_access_control.erl

```diff
@@ -80,41 +68,32 @@ auth(Client, Password, [{Mod, State, _Seq} | Mods]) ->
         {'EXIT', Error} -> {error, Error}
     end.
 
-%%------------------------------------------------------------------------------
 %% @doc Check ACL
-%% @end
-%%------------------------------------------------------------------------------
 -spec check_acl(Client, PubSub, Topic) -> allow | deny when
       Client :: mqtt_client(),
       PubSub :: pubsub(),
       Topic  :: binary().
 check_acl(Client, PubSub, Topic) when ?IS_PUBSUB(PubSub) ->
     case lookup_mods(acl) of
-        [] -> allow;
+        []      -> allow;
         AclMods -> check_acl(Client, PubSub, Topic, AclMods)
     end.
 check_acl(#mqtt_client{client_id = ClientId}, PubSub, Topic, []) ->
-    lager:error("ACL: nomatch when ~s ~s ~s", [ClientId, PubSub, Topic]),
+    lager:error("ACL: nomatch for ~s ~s ~s", [ClientId, PubSub, Topic]),
     allow;
-check_acl(Client, PubSub, Topic, [{M, State, _Seq}|AclMods]) ->
-    case M:check_acl({Client, PubSub, Topic}, State) of
+check_acl(Client, PubSub, Topic, [{Mod, State, _Seq}|AclMods]) ->
+    case Mod:check_acl({Client, PubSub, Topic}, State) of
```

Comment:
Remove the dynamic function call on line 85. Only modules that define callbacks should make dynamic calls.